src/pymor/parameters/functionals.py

```diff
@@ -42,6 +42,14 @@ def evaluate(self, mu=None):
         mu = self.parse_parameter(mu)
         return mu[self.component_name].item(self.coordinates)
 
+    def d_mu(self, component, index=()):
+        if not isinstance(index, tuple):
+            index = (index,)
```

Comment:
I think this is a bit dangerous: if I were to pass a list instead of a tuple then this code would lead to always return a zero `ParameterFunctional` instead of, e.g., raising. Maybe a better way would be:
1. If index is a `Number`, convert to tuple of length one.
2. Convert `index` to a tuple by calling `tuple(index)`.
3. Check that every element of `index` is a `Number`.
4. Check that all indices are within the allowed range in case `component` is contained in the `parameter_type`.